.github/CONTRIBUTING.md

```diff
@@ -1,17 +1,28 @@
-# Contributing Guidelines
+# Contribution Guidelines
 
 ## Contents
 
-* [General guidelines](#general-guidelines)
-* [C++ coding guidelines](#c-coding-guidelines)
-* [Pull requests](#pull-requests)
+* [Getting started](#getting-started)
+* [Writing code](#writing-code)
+* [Adding unit tests](#adding-unit-tests)
+* [Opening a pull request](#opening-a-pull-request)
 * [Code of conduct](#code-of-conduct)
 
-## General guidelines
+## Getting started
 
-* When contributing to this repository, please first inform or discuss the
-  change(s) you wish to make via an issue. This helps in letting others know
-  what you're working on.
+* Start by perusing the repository. Take a look at how the algorithms are implemented, how tests
+  are written, and which files go where. 
+
+* Read the [contribution guidelines](../C++/CONTRIBUTING.md).
+
+* Before beginning your contribution, [create an issue][issue-guide]. In your issue's 
+  description, please describe the addition or change you wish to make. This helps us guide 
+  your contribution, and it lets others know what you're working on.
+
```

Comment:
I think it would be better to move the first three steps from the "Opening a pull request" section here, as they are more related to the "getting started" process.